src/components/PostDetails.tsx

```diff
@@ -1,106 +1,133 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import * as serverAction from '../api/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isCreateComment, setIsCreateComment] = useState(false);
+
+  useEffect(() => {
+    if (post) {
+      setIsCreateComment(false);
+      setIsLoading(true);
+
+      serverAction
+        .getComments(post.id)
+        .then(result => {
+          const results = result.map(comment => ({
+            id: comment.id,
+            postId: comment.postId,
+            name: comment.name,
+            body: comment.body,
+            email: comment.email,
+          }));
+
+          setComments(results);
+        })
+        .catch(() => setIsError(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [post]);
+
+  const deleteComment = (commentId: number) => {
+    setComments(prev => prev.filter(comment => comment.id !== commentId));
+    serverAction.deleteComment(commentId);
```

Comment:
Consider adding error handling for the `deleteComment` API call. If the deletion fails, the comment will still be removed from the UI, which might lead to inconsistencies between the client and server states.